app/models/rental.rb

```diff
@@ -1,4 +1,10 @@
 class Rental < ApplicationRecord
   belongs_to :product
   belongs_to :user
+
+  validates :begin_date, presence: true
+  validates :end_date, presence: true
```

Comment:
I ll have to study how to do that, but I think that "end_date > begin_date" is the best validation, so the the rental dates will make sense and rental_price will be never negative